src/galaxy.cpp

```diff
@@ -722,6 +733,179 @@ void SelectPlayerView::clickPlayer(int x, int y, int arg) {
 	exitView();
 }
 
+PlanetsListView::PlanetsListView(const GameState *game) : _game(game) {
+	Widget *w = NULL;
+	const uint8_t *pal;
+
+	_bg = gameAssets->getImage(PLANET_ARCHIVE, ASSET_PLANETLIST_BG);
+	_planetAssets = gameAssets->getImage(COLONY_VIEW_PLANET_LIST_ARCHIVE, COLONY_VIEW_PLANET_LIST_PALETTE);
+
+	pal = _bg->palette();
+
+	w = createWidget(453, 440, 153, 18);
+	w->setMouseUpCallback(MBUTTON_LEFT,
+		GuiMethod(*this, &PlanetsListView::clickReturn));
+	w->setClickSprite(MBUTTON_LEFT, PLANET_ARCHIVE, ASSET_PLANETLIST_RETURN_BUTTON,
+		pal, 1);
+}
+
+void PlanetsListView::redraw(unsigned curtick) {
+	Font *fnt, *smallFnt;
+	const uint8_t *pal;
+	int x, y, displayedPlanet = 1;
+	char buf[32];
+	const int climateOffset = 5,
+			  mineralsOffset = 5,
+			  sizeOffset = 5,
+			  nameOffset = 10,
+			  rowHeight = 55,
+			  planetListMaxItems = 8,
+			  offset = 10;
+	const char *romanNumbers[5] = { "I", "II", "III", "IV", "V"};
+	unsigned color = FONT_COLOR_PLANET_LIST, asset_id = 0;
+
+	clearScreen();
+	fnt = gameFonts->getFont(FONTSIZE_SMALL);
+	smallFnt = gameFonts->getFont(FONTSIZE_SMALLER);
+
+	for (auto planet: _game->_planets) {
+		if (planet.star > 0 && _game->_colonies[planet.colony].owner) {
+			sprintf(buf, "%s %s", _game->_starSystems[planet.star].name, romanNumbers[planet.orbit]);
+
+			// Planet climate
+			switch(planet.climate) {
+			case TOXIC:
+				sprintf(buf, "%s", "Toxic");
```

Comment:
You can load all these strings using `gameLang->estrings()`. Although you'll need a string ID mapping because some of the strings are out of order in the estrings file...